pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -527,6 +528,7 @@ func (configgen *ConfigGeneratorImpl) buildSidecarInboundListenerForPortOrUDS(li
 				},
 				ServerName: EnvoyServerName,
 			},
+			addInboundHealthCheckFilter: true,
```

Comment:
Same comment - flag should control if this is on/off ( or env so it works with current installers ).

Please add lots of comments on the flag -and/or in the PR - on the impact and behavior changes.

Reply:
@liamawhite can you doublecheck if tracing still works when you do this? I remember @objectiser adding a feature to envoy to disable tracing when the call goes through a health check filter. And we have a pilot plugin that adds it to management Ports that have HTTP probes. So, adding the health check by default for all inbound could cause tracing issues. @objectiser to confirm